web/web.go

```diff
@@ -0,0 +1,1022 @@
+	reqBody, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Body Read Error : %v", err), http.StatusInternalServerError)
+		return
+	}
+	var dc dumpConfig
+	err = json.Unmarshal(reqBody, &dc)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Request Body parse error : %v", err), http.StatusBadRequest)
+		return
+	}
+	f, err := os.Open(dc.FilePath)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("failed to open dump file %v : %v", dc.FilePath, err), http.StatusNotFound)
+		return
+	}
+	conv, err := conversion.SchemaConv(dc.Driver, &conversion.IOStreams{In: f, Out: os.Stdout}, 0)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Schema Conversion Error : %v", err), http.StatusNotFound)
+		return
+	}
+	app.conv = conv
+	app.driver = dc.Driver
+	app.dbName = ""
+	app.sessionFile = ""
+	app.sourceDB = nil
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(conv)
+}
+
+// getDDL returns the Spanner DDL for each table in alphabetical order.
+// Unlike internal/convert.go's GetDDL, it does not print tables in a way that
+// respects the parent/child ordering of interleaved tables, also foreign keys
+// and secondary indexes are skipped. This means that getDDL cannot be used to
+// build DDL to send to Spanner.
+func getDDL(w http.ResponseWriter, r *http.Request) {
+	c := ddl.Config{Comments: true, ProtectIds: false}
+	var tables []string
+	for t := range app.conv.SpSchema {
+		tables = append(tables, t)
+	}
+	sort.Strings(tables)
+	ddl := make(map[string]string)
+	for _, t := range tables {
+		ddl[t] = app.conv.SpSchema[t].PrintCreateTable(c)
+	}
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(ddl)
+}
+
+// getSummary returns table wise summary of conversion.
+func getSummary(w http.ResponseWriter, r *http.Request) {
+	reports := internal.AnalyzeTables(app.conv, nil)
+	summary := make(map[string]string)
+	for _, t := range reports {
+		var body strings.Builder
+		for _, x := range t.Body {
+			body.WriteString(x.Heading + "\n")
+			for i, l := range x.Lines {
+				body.WriteString(fmt.Sprintf("%d) %s.\n\n", i+1, l))
+			}
+		}
+		summary[t.SrcTable] = body.String()
+	}
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(summary)
+}
+
+// getOverview returns the overview of conversion.
+func getOverview(w http.ResponseWriter, r *http.Request) {
+	var buf bytes.Buffer
+	bufWriter := bufio.NewWriter(&buf)
+	internal.GenerateReport(app.driver, app.conv, bufWriter, nil, false, false)
+	bufWriter.Flush()
+	overview := buf.String()
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(overview)
+}
+
+// getTypeMap returns the source to Spanner typemap only for the
+// source types used in current conversion.
+func getTypeMap(w http.ResponseWriter, r *http.Request) {
+	if app.conv == nil || app.driver == "" {
+		http.Error(w, fmt.Sprintf("Schema is not converted or Driver is not configured properly. Please retry converting the database to spanner."), http.StatusNotFound)
```

Comment:
s/spanner/Spanner/ in error message.

Reply:
changed to 
Schema is not converted or Driver is not configured properly. Please retry converting the database to Spanner.